### PR TITLE
Radar.py with object vector definition docstring

### PR DESCRIPTION
**Pull request by hobson**

### CODE REVIEW

File: object_detection/utils/radar.py [hunk 1]
```diff
@@ -1,21 +1,92 @@
...
+                self.samples.append([])
+        else:
+            for i, row in enumerate(samples):
+                # list of lists of dicts, lists, Series where each element is the "state" of a detected object
+                self.samples += [pd.DataFrame()]
+                for j, obj in enumerate(row):
+                    self.samples.append(pd.Series(list(zip(*row))[1], index=list(zip(*row))[0], name=i))
         self.now = 0
 
+    def update_state(self, boxes, classes, scores, category_index=None, window=10, max_boxes_to_draw=None, min_score_thresh=.4):
```
Comment (ashwinkannan94) [inline object_detection/utils/radar.py]: The minimum score threshold is set to 0.4, is that arbitrarily set for now or is that what is going to be for always?

File: object_detection/utils/radar.py [hunk 2]
```diff
@@ -1,21 +1,92 @@
...
+        """ Revise state based on latest frame of information (object boxes)
+
+        Args:
+            boxes (list): 2D numpy array of shape (N, 4): (ymin, xmin, ymax, xmax), in normalized format between [0, 1].
+            classes,
+        Args (thats hould be class attributes):
```
Comment (ashwinkannan94) [inline object_detection/utils/radar.py]: Args( that should)*

File: object_detection/utils/radar.py [hunk 3]
```diff
@@ -1,21 +1,92 @@
...
+            category_index (dict of dicts): {1: {'id': 1, 'name': 'person'}, 2: {'id': 2, 'name': 'bicycle'},...}
+
+        """
+        category_index = self.category_index if category_index is None else category_index
+        num_boxes = min([boxes.shape[0] if max_boxes_to_draw is None else max_boxes_to_draw, boxes.shape[0], len(classes)])
+
+        # FIXME: unify self.update_state.states with self.samples
+        if self.update_states is None:
```
Comment (ashwinkannan94) [inline object_detection/utils/radar.py]: There is no update_states in the function parameters. Either need to update the function or this does thing.

File: object_detection/utils/radar.py [hunk 4]
```diff
@@ -1,21 +1,92 @@
...
+            # Initialize a matrix of state vectors for the past 20 frames
+            self.update_state.row = 0
+            self.update_state.window = 20
+            self.update_state.columns = pd.DataFrame(list(self.category_index.values())).set_index('id', drop=True)['name']
+            self.update_state.states = pd.DataFrame(pd.np.zeros((20, len(self.category_index)),
+                                                                dtype=int), columns=self.update_state.columns)
+            self.update_state.state0 = pd.Series(index=self.update_state.columns)
```
Comment (ashwinkannan94) [inline object_detection/utils/radar.py]: There is no update_state function parameter being passed in. This few lines above are not going to update anything.

File: object_detection/utils/radar.py [hunk 5]
```diff
@@ -1,21 +1,95 @@
...
+        """
+        category_index = self.category_index if category_index is None else category_index
+        num_boxes = min([boxes.shape[0] if max_boxes_to_draw is None else max_boxes_to_draw,
+                         boxes.shape[0], len(classes)])
+
+        # FIXME: unify self.update_state.states with self.samples
+        if self.update_state.states is None:
+            # Initialize a matrix of state vectors for the past 20 frames
+            self.update_state.row = 0
+            self.update_state.window = 20
```
Comment (ashwinkannan94) [inline object_detection/utils/radar.py]: The window has been set to 20 but on the function definition, the window is set to 10.

File: object_detection/utils/radar.py [hunk 6]
```diff
@@ -1,21 +1,96 @@
...
+      ['red', 5], ['blue', 10], ['yellow', 2],
+      ['purple', 6], ['orange', 1], ['green', 1],
+    ]
+    -1 < x < 1 where 0 is the center of the frame, + 1 is far right
+    -1 < y < 1 where 0 is center and +1 is the top of the frame
+    z is TBD
+    width and height are in the same scale as x, y, e.g. width = 2.0 / pixel_width
+    depth units is TBD
+    colors are pixel counts within a portion of the bounding box near the center
+    """
```
Comment (alxrsngrtn) [inline object_detection/utils/radar.py]: Docstring should adhere to contributing guidelines. Need args + returns at least.

**Pull request merged**